scripts/includes/common.sh

```diff
@@ -0,0 +1,248 @@
+  if (! id stanley 2>/dev/null); then
+    sudo useradd stanley
+  fi
+
+  sudo mkdir -p /home/stanley/.ssh
+
+  # Generate ssh keys on StackStorm box and copy over public key into remote box.
+  sudo ssh-keygen -f /home/stanley/.ssh/stanley_rsa -P ""
+
+  # Authorize key-base access
+  sudo sh -c 'cat /home/stanley/.ssh/stanley_rsa.pub >> /home/stanley/.ssh/authorized_keys'
+  sudo chmod 0600 /home/stanley/.ssh/authorized_keys
+  sudo chmod 0700 /home/stanley/.ssh
+  sudo chown -R stanley:stanley /home/stanley
+
+  # Enable passwordless sudo
+  sudo sh -c 'echo "stanley    ALL=(ALL)       NOPASSWD: SETENV: ALL" >> /etc/sudoers.d/st2'
+  sudo chmod 0440 /etc/sudoers.d/st2
+
+  # Disable requiretty for all users
+  sudo sed -i -r "s/^Defaults\s+\+?requiretty/# Defaults requiretty/g" /etc/sudoers
+}
+
+
+configure_st2_cli_config() {
+  # Configure CLI config (write credentials for the root user and user which ran the script)
+  ROOT_USER="root"
+  CURRENT_USER=$(whoami)
+
+  : "${HOME:=`eval echo ~$(whoami)`}"
+
+  ROOT_USER_CLI_CONFIG_DIRECTORY="/root/.st2"
+  ROOT_USER_CLI_CONFIG_PATH="${ROOT_USER_CLI_CONFIG_DIRECTORY}/config"
+
+  CURRENT_USER_CLI_CONFIG_DIRECTORY="${HOME}/.st2"
+  CURRENT_USER_CLI_CONFIG_PATH="${CURRENT_USER_CLI_CONFIG_DIRECTORY}/config"
+
+  if [ ! -d ${ROOT_USER_CLI_CONFIG_DIRECTORY} ]; then
+    sudo mkdir -p ${ROOT_USER_CLI_CONFIG_DIRECTORY}
+  fi
+
+  sudo sh -c "cat <<EOT > ${ROOT_USER_CLI_CONFIG_PATH}
+[credentials]
+username = ${USERNAME}
+password = ${PASSWORD}
+EOT"
+
+  # Write config for root user
+  if [ "${CURRENT_USER}" == "${ROOT_USER}" ]; then
+      return
+  fi
+
+  # Write config for current user (in case current user != root)
+  if [ ! -d ${CURRENT_USER_CLI_CONFIG_DIRECTORY} ]; then
+    sudo mkdir -p ${CURRENT_USER_CLI_CONFIG_DIRECTORY}
+  fi
+
+  sudo sh -c "cat <<EOT > ${CURRENT_USER_CLI_CONFIG_PATH}
+[credentials]
+username = ${USERNAME}
+password = ${PASSWORD}
+EOT"
+
+  # Fix the permissions
+  sudo chown -R ${CURRENT_USER}:${CURRENT_USER} ${CURRENT_USER_CLI_CONFIG_DIRECTORY}
+}
+
+
+generate_symmetric_crypto_key_for_datastore() {
+  DATASTORE_ENCRYPTION_KEYS_DIRECTORY="/etc/st2/keys"
+  DATASTORE_ENCRYPTION_KEY_PATH="${DATASTORE_ENCRYPTION_KEYS_DIRECTORY}/datastore_key.json"
+
+  sudo mkdir -p ${DATASTORE_ENCRYPTION_KEYS_DIRECTORY}
+  sudo st2-generate-symmetric-crypto-key --key-path ${DATASTORE_ENCRYPTION_KEY_PATH}
+
+  # Make sure only st2 user can read the file
+  sudo usermod -a -G st2 st2
+  sudo chgrp st2 ${DATASTORE_ENCRYPTION_KEYS_DIRECTORY}
+  sudo chmod o-r ${DATASTORE_ENCRYPTION_KEYS_DIRECTORY}
+  sudo chgrp st2 ${DATASTORE_ENCRYPTION_KEY_PATH}
+  sudo chmod o-r ${DATASTORE_ENCRYPTION_KEY_PATH}
+
+  # set path to the key file in the config
+  sudo crudini --set /etc/st2/st2.conf keyvalue encryption_key_path ${DATASTORE_ENCRYPTION_KEY_PATH}
+
+  sudo st2ctl restart-component st2api
+}
+
+
+verify_st2() {
+  st2 --version
+  st2 -h
+
+  st2 auth $USERNAME -p $PASSWORD
```

Comment:
Should we replace `st2 auth` with `st2 login` ?

Reply:
I am inclined to say yes, we should move to to `st2 login`...but since this PR is already pretty big, maybe it should mostly just be "a better way of generating the existing content", and then we do future PRs to improve the actual content?

Reply:
Since this PR is all about consolidating common code and auto generating the bootstrap scripts, and we're going to be making many other small changes to these scripts.. I vote that we handle this change in a subsequent PR.

Reply:
I agree - those changes should be made in another PR once this one is merged.